lucet-runtime/lucet-runtime-internals/src/instance/signals.rs

```diff
@@ -180,31 +180,48 @@ extern "C" fn handle_signal(signum: c_int, siginfo_ptr: *mut siginfo_t, ucontext
                 true
             }
             SignalBehavior::Default => {
-                // safety: pointer is checked for null at the top of the function, and the
-                // manpage guarantees that a siginfo_t will be passed as the second argument
-                let siginfo = unsafe { *siginfo_ptr };
-                let rip_addr = rip as usize;
-                // If the trap table lookup returned unknown, it is a fatal error
-                let unknown_fault = trapcode.is_none();
-
-                // If the trap was a segv or bus fault and the addressed memory was outside the
-                // guard pages, it is also a fatal error
-                let outside_guard = (siginfo.si_signo == SIGSEGV || siginfo.si_signo == SIGBUS)
-                    && !inst.alloc.addr_in_heap_guard(siginfo.si_addr_ext());
-
-                // record the fault and jump back to the host context
-                inst.state = State::Faulted {
-                    details: FaultDetails {
-                        fatal: unknown_fault || outside_guard,
-                        trapcode: trapcode,
-                        rip_addr,
-                        // Details set to `None` here: have to wait until `verify_trap_safety` to
-                        // fill in these details, because access may not be signal safe.
-                        rip_addr_details: None,
-                    },
-                    siginfo,
-                    context: ctx.into(),
+                /*
+                 * /!\ WARNING: LOAD-BEARING THUNK /!\
```

Comment:
I'm at a loss of how to test paths through our signal handler, just putting that out here. A more specialized version of `sigsegv_handler_during_guest`? Setting the host thread's signal stack to `SIGSTKSZ + page`, testing that we don't fault with only `SIGSTKSZ`, and if we do fault, letting `panic!` do its thing with slightly more breathing room?

I think I just need to try writing such a test.

Reply:
That's a tricky one; maybe mmap yourself a `SIGSTK + page` as you mentioned, run the test, then assert that the lowermost page is all zeroes? Wouldn't be perfect, but unless we're explicitly covering our tracks after running the handler it should give us some assurance.